src/main/java/apoc/path/PathExplorer.java

```diff
@@ -95,16 +98,30 @@ private Uniqueness getUniqueness(String uniqueness) {
 			, String pathFilter
 			, String labelFilter
 			, long minLevel
-			, long maxLevel, boolean bfs, Uniqueness uniqueness) {
+			, long maxLevel, boolean bfs, Uniqueness uniqueness, boolean optionalMatch) {
 		// LabelFilter
 		// -|Label|:Label|:Label excluded label list
 		// +:Label or :Label include labels
 
-		Traverser traverser = traverse(db.traversalDescription(), startNodes, pathFilter, labelFilter, minLevel, maxLevel, uniqueness,bfs);
-		return traverser.stream().map( PathResult::new );
+		Traverser traverser = traverse(db.traversalDescription(), startNodes, pathFilter, labelFilter, minLevel, maxLevel, uniqueness,bfs, optionalMatch);
+	    
+		Stream<PathResult> returnStream = traverser.stream().map( PathResult::new );
+		if (optionalMatch) {
```

Comment:
can we simplify this to test on traverser `if (!optional || traverser.hasNext())` and then return the regular stream?